source/lib/vagrant-openstack-provider/client/nova.rb

```diff
@@ -51,7 +51,16 @@ def create_server(env, options)
         server = {}.tap do |s|
           s['name'] = options[:name]
           if options[:image_ref].nil?
-            s['block_device_mapping'] = [{ volume_id: options[:volume_boot][:id], device_name: options[:volume_boot][:device] }]
+            s['block_device_mapping'] = [{ volume_id: options[:volume_boot][:id],
+                                           device_name: options[:volume_boot][:device] }] unless options[:volume_boot][:id].nil?
+            s['block_device_mapping_v2'] = [{ boot_index: '0',
+                                              volume_size: options[:volume_boot][:size],
+                                              uuid: options[:volume_boot][:image],
+                                              device_name: options[:volume_boot][:device],
+                                              source_type: 'image',
+                                              destination_type: 'volume',
+                                              delete_on_termination: options[:volume_boot][:delete_on_destroy] }]\
+                                              unless options[:volume_boot][:image].nil?
```

Comment:
With the change i suggest in `config_resolver.rb`, here it will become

```ruby
s['block_device_mapping'] = ... if options[:volume_boot].key?(:id)
s['block_device_mapping_v2'] = ... if options[:volume_boot].key?(:image)
```